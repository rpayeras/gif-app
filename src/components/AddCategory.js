import React, { useState } from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({onSubmit}) => {
    const [inputText, setInputText] = useState('')

    const handleInput = (e) => setInputText(e.target.value);

    const handleSubmit = (e) => {
        e.preventDefault()

        if(inputText.trim().length > 2){
            onSubmit(inputText)
            setInputText('')
        }
    }

    return (
        <>
            <form onSubmit={handleSubmit}>
                <input name="name" type="text" value={inputText} onChange={handleInput} placeholder="Add a category..."/>
            </form>
        </>
    );
};

AddCategory.propTypes = {
    onSubmit: PropTypes.func.isRequired
}