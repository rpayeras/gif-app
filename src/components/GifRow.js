import React, { useState, useEffect } from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifRow = ({category}) => {
  const {data: images, loading} = useFetchGifs(category)

  return (
    <>
      <h3>{category}</h3>
      {loading ?? <p>'Loading...'</p>}
      { <div className='card-grid '>
        {images.map((img) => {
          return (
            <GifGridItem {...img} key={img.id} />
          )
        })}
    </div> }
    </>
  );
};

GifRow.propTypes = {
  category: PropTypes.string.isRequired
}