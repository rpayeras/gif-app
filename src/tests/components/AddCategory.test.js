import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"

describe('Testing with AddCategory component', () => {
    const setCategories = jest.fn()
    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks()
        wrapper = shallow(<AddCategory onSubmit={setCategories}/>)
    })

    test('should render correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('input must change', () => {
        const input = wrapper.find('input');
        input.simulate('change', {target:{value:'test1'}})
        
        expect(wrapper.find('input').prop('value')).toBe('test1')
    })

    test('not should call on submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled()
    })

    test('should call setCategories and clean input', () => {
        wrapper.find('input').simulate('change', { target: { value:'test1' }})
        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect(setCategories).toHaveBeenCalled()
        expect(wrapper.find('input').prop('value')).toBe('')
    })
})