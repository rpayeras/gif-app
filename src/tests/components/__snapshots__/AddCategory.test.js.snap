// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing with AddCategory component should render correctly 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Add a category..."
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
